libzeth/tests/mpc/groth16/mpc_test.cpp

```diff
@@ -71,33 +71,38 @@ srs_mpc_phase2_accumulator<ppT> dummy_initial_accumulator(
 
 TEST(MPCTests, HashToG2)
 {
+    // Data here is specialised for alt_bn128. Call init_public_params on it,
+    // just in case `ppT` is set to a different curve.
+    using pp = libff::alt_bn128_pp;
```

Comment:
Why not use `ppT` which is defined as `using ppT = libff::alt_bn128_pp;` at the top of the file?

Reply:
Thanks.  `ppT` is supposed to represent the pairing used to run the tests, *except* those that are specialized for a specific pairing.  So `ppT` is the default (eventually it would be nice to parameterize all these tests by ppT so we can run over all pairings - there should be no reason to restrict to the default - but I didn't plan to address that here), the second call to `init_params` in `main` is just in case `ppT` is not set to `alt_bn_128`.

The `pp` is kind of independent - it's just local to the function for convenience.  If we expand everything to `libff::alt_bn128_pp` everywhere its many extra lines of code for no extra clarity.

Hope it's clear in the new version.